lib/Sema/CSSyntacticElement.cpp

```diff
@@ -905,6 +905,38 @@ class SyntacticElementConstraintGenerator
   }
 
   void visitBraceStmt(BraceStmt *braceStmt) {
+    auto &ctx = cs.getASTContext();
+
+    auto addResultDefault = [&](ClosureExpr *closure) {
+      if (context.getBody() == braceStmt) {
```

Comment:
Nit: this should be `closure->getBody() == braceStmt` and you can add that as another condition to the check for `hasExplicitResult`. I am also wondering where a better check here would be `locator->directlyAt<ClosureExpr>()` because in case of result builder transform the transformed body is disconnected from the closure expression and `getBody` points to the original untranformed one…

Reply:
I can try using `locator->directlyAt<ClosureExpr>()` just to see

Reply:
Sounds good!